app/controllers/categorisation/CyaCategorisationController.scala

```diff
@@ -156,43 +155,49 @@ class CyaCategorisationController @Inject() (
 
   def onSubmit(recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      CategoryRecord.build(request.userAnswers, request.eori, recordId, categorisationService) match {
-        case Right(categoryRecord) =>
-          auditService.auditFinishCategorisation(
+      goodsRecordConnector.getRecord(request.eori, recordId).flatMap { oldRecord =>
+        CategoryRecord
+          .build(
+            request.userAnswers,
             request.eori,
-            request.affinityGroup,
             recordId,
-            categoryRecord
-          )
-
-          val result = for {
-            oldRecord <- goodsRecordConnector.getRecord(request.eori, recordId)
-            _         <-
-              goodsRecordConnector
-                .updateCategoryAndComcodeForGoodsRecord(request.eori, recordId, categoryRecord, oldRecord)
-          } yield {
-            dataCleansingService.deleteMongoData(request.userAnswers.id, CategorisationJourney)
-            Redirect(
-              navigator.nextPage(
-                CyaCategorisationPage(recordId),
-                NormalMode,
-                request.userAnswers
-              )
+            categorisationService
+          ) match {
+          case Right(categoryRecord) =>
+            auditService.auditFinishCategorisation(
+              request.eori,
+              request.affinityGroup,
+              recordId,
+              categoryRecord
```

Comment:
Would it make sense for this to be in the below for comp as well? Or is the intention here to ensure we don't fall over if connection to CIP fails?